### PR TITLE
implement basic auto format logic

### PR DESCRIPTION
**Pull request by cmac1000**
For discussion, mostly: we should still decide if we intend to support this kind of logic, how it would interact with, say, Accept headers and browser capabilities.

This could probably use some profiling too, not sure what the best approach is to ensure that the performance is good enough (pretty snappy locally though, and caching will do a lot in practice)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -160,7 +160,7 @@ Format returns the input image in the specified type, rather than being left as
 Where:
 
 - format -- indicates that ITS should perform a format transform
-- ext -- the requested output type. Can only be `jpg`, `png` or `webp`
+- ext -- the requested output type. Can only be `jpg`, `png`, `webp` or `auto` (`auto` will examine the issue and return either a jpg or png depending on the image's complexity)
```
Comment (wamberg) [inline README.md]: "issue" = "image"

File: its/optimize.py [hunk 2]
```diff
@@ -16,10 +16,27 @@
 ImageFile.MAXBLOCK = 2 ** 20  # for JPG progressive saving
 
 
+def identify_best_format(img: Image.Image) -> str:
+    identical_pixel_threshold = 0.001
+    common_values_threshold = 50
+    hist = img.histogram()
+    total = sum(hist)
+    common_vals = [v / total for v in hist if v / total > identical_pixel_threshold]
```
Comment (wamberg) [inline its/optimize.py]: Could this be:

```python
common_vals = [v for v in hist if v / total > identical_pixel_threshold]
```

We don't do anything with `common_vals` other than run it through `len()`. Think how many picoseconds we could  save by avoiding the `v / total` calculation!!

Comment (cmac1000) [reply its/optimize.py]: 👍 1fc1be622

File: README.md (hunk 1 above)
Comment (cmac1000) [reply README.md]: 👍 29a3e560

**Pull request merged**